cilium/network_filter.cc

```diff
@@ -162,6 +162,8 @@ Network::FilterStatus Instance::onNewConnection() {
       ENVOY_CONN_LOG(warn, "cilium.network (egress): No destination address", conn);
       return false;
     }
+
+    policy_fs->dst_address = dst_address;
```

Comment:
Isn't it possible to extract the correct dst address when building the policy filterstate? 

i'm asking because previous PRs tried to create dedicated fitlerstates and socketoptions. I kind of fear that we're on the way of re-adding more and more properties/functionality to the existing "components" instead of adding new one.

alternatively we could create a new filterstate that is only responsible to transport this information.

Reply:
having a dedicated filterstate is awesome point, I don't need to mess with readonly or mutable type for existing policy fs, thanks :+1: